app/controllers/admin/employees_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Admin
+  class EmployeesController < AdminController
+    def index
+      render :index, locals: { employees: Employee.includes(:orders).all }
+    end
+
+    def show
+      render :show, locals: { employee: employee }
+    end
+
+    def edit
+      render :edit, locals: { employee: employee }
+    end
+
+    def update
+      if employee.update(employee_params)
+        redirect_to admin_employees_path(employee), notice: 'Employee was successfully updated.'
+      else
+        render :edit, locals: { employee: employee }
+      end
+    end
+
+    def destroy
+      employee.destroy
+      redirect_to admin_employees_url, notice: 'Employee was successfully destroyed.'
+    end
+
+    def edit_kudos_for_all
+      render :update_kudos_for_all, locals: { employees: Employee.all }
+    end
+
+    def update_kudos_for_all
+      if add_kudos_param >= 1 && add_kudos_param <= 20
+        Employee.all.each do |employee|
```

Comment:
what if the operation would stop in the middle you'd end up with few updated users and few not updated users :). Have a look at the db transactions :)

Reply:
We're calling the transaction method on the ActiveRecord::Base class and passing it a block. Every database operation that happens inside that block will be sent to the database as a transaction.

If any kind of unhandled error happens inside the block, the transaction will be aborted, and no changes will be made to the DB.

```
ActiveRecord::Base.transaction do
            Employee.find_each do |employee|
              employee.number_of_available_kudos += add_kudos_param
              employee.save!
            end
end
```